dockstore-client/src/main/java/io/dockstore/client/cli/nested/ApiClientExtended.java

```diff
@@ -16,16 +17,55 @@
 
 public class ApiClientExtended extends ApiClient {
 
+    /**
+     *
+     * @param key
+     * @return
+     */
+    private Optional<MediaType> getMediaType(Optional<String> key) {
+        // if key happens to be null or we don't recognize the key then return null
+        Optional<MediaType> mediaType = Optional.empty();
+        if (key.isPresent()) {
+            String keyWESParam = key.get().toLowerCase();
+            switch (keyWESParam) {
+            case "workflow_params":
+            case "tags":
+            case "workflow_engine_parameters":
+                mediaType =  Optional.of(MediaType.APPLICATION_JSON_TYPE);
+                break;
+            case "workflow_attachment":
+                mediaType =  Optional.of(MediaType.APPLICATION_OCTET_STREAM_TYPE);
+                break;
+            case "workflow_url":
+            case "workflow_type":
+            case "workflow_type_version":
+                mediaType =  Optional.of(MediaType.TEXT_PLAIN_TYPE);
+                break;
+            default:
+            }
+        }
+        return mediaType;
+    }
+
 
     public void createBodyPart(MultiPart multiPart, String key, Object formObject) {
+        Optional<MediaType> mediaType = getMediaType(Optional.ofNullable(key));
         if (formObject instanceof File) {
             File file = (File)formObject;
             FormDataContentDisposition contentDisp = FormDataContentDisposition.name(key)
                     .fileName(file.getName()).size(file.length()).build();
-            multiPart.bodyPart(new FormDataBodyPart(contentDisp, file, MediaType.APPLICATION_OCTET_STREAM_TYPE));
+            if (mediaType.isPresent()) {
+                multiPart.bodyPart(new FormDataBodyPart(contentDisp, file, mediaType.get()));
+            } else {
+                multiPart.bodyPart(new FormDataBodyPart(contentDisp, file, MediaType.APPLICATION_OCTET_STREAM_TYPE));
```

Comment:
mediaType.orElseGet(MediaType.APPLICATION_OCTET_STREAM_TYPE);

Reply:
This is even better

Reply:
Used the suggestion and fixed code